src/watertap_contrib/reflo/solar_models/surrogate/flat_plate/flat_plate_surrogate.py

```diff
@@ -0,0 +1,305 @@
+            doc="Storage volume for flat plate system",
+        )
+
+        self.heat_load = Var(
+            initialize=500,
+            bounds=[100, 1000],
+            units=pyunits.MW,
+            doc="Rated plant heat capacity in MWt",
+        )
+        self.hours_storage = Var(
+            initialize=20,
+            bounds=[0, 26],
+            units=pyunits.hour,
+            doc="Rated plant hours of storage",
+        )
+        self.temperature_hot = Var(
+            initialize=70,
+            bounds=[50, 100],
+            units=pyunits.K,
+            doc="Hot outlet temperature",
+        )
+
+        self.heat_annual = Var(
+            initialize=1000,
+            units=pyunits.kWh,
+            doc="Annual heat generated by flat plate",
+        )
+        self.electricity_annual = Var(
+            initialize=20,
+            units=pyunits.kWh,
+            doc="Annual electricity consumed by flat plate",
+        )
+
+        self.surrogate_inputs = [
+            self.heat_load,
+            self.hours_storage,
+            self.temperature_hot,
+        ]
+        self.surrogate_outputs = [self.heat_annual, self.electricity_annual]
+
+        self.input_labels = ["heat_load", "hours_storage", "temperature_hot"]
+        self.output_labels = ["heat_annual", "electricity_annual"]
+
+        if self.config.surrogate_model_file:
+            self.surrogate_file = os.path.join(
+                os.path.dirname(__file__), self.config.surrogate_model_file
+            )
+
+        else:
+            self.dataset_filename = os.path.join(
+                os.path.dirname(__file__), "data/flat_plate_data.pkl"
+            )
+            self.surrogate_file = os.path.join(
+                os.path.dirname(__file__), "flat_plate_surrogate.json"
+            )
+
+        self._load_surrogate()
+
+        self.heat_constraint = Constraint(
+            expr=self.heat_annual
+            == self.heat * pyunits.convert(1 * pyunits.year, to_units=pyunits.hour)
+        )
+
+        self.electricity_constraint = Constraint(
+            expr=self.electricity_annual
+            == self.electricity
+            * pyunits.convert(1 * pyunits.year, to_units=pyunits.hour)
+        )
+
+        self.collector_area_total_constraint = Constraint(
+            expr=self.collector_area_total
+            * (self.FR_ta - self.FR_UL * self.factor_delta_T)
+            == pyunits.convert(self.heat_load, to_units=pyunits.kilowatt)
+        )
+
+        self.number_collectors_constraint = Constraint(
+            expr=self.number_collectors
+            == self.collector_area_total / self.collector_area_per
+        )
+
+        self.storage_volume_constraint = Constraint(
+            expr=self.storage_volume
+            == pyunits.convert(
+                (
+                    (self.hours_storage * self.heat_load)
+                    / (
+                        self.specific_heat_water
+                        * self.temperature_cold
```

Comment:
I might be wrong but I think we should use temp dif to calculate the energy available from the heat transfer fluid, as Mukta did in the TES model:
`* self.temperature_cold` --> `* (self.temp_design - self.temperature_cold)`